pkg/kubernetes/cluster.go

```diff
@@ -196,13 +230,20 @@ func setClusterCNI(config map[string]any) {
 	config[cniKey] = cniDefaultValue
 }
 
-func setClusterAPIAddress(config map[string]any, apiAddress string, port int) {
-	if apiAddress == "" {
+func setClusterAPIAddress(config map[string]any, ip4 *netip.Addr, ip6 *netip.Addr, port uint16, prioritizeIPv6 bool) {
+	if ip4 == nil && ip6 == nil {
 		zap.S().Warn("Attempted to set an empty cluster API address")
```

Comment:
Let's panic() instead -- we have a graceful handling for it and we shouldn't proceed anyway.